src/actions.rs

```diff
@@ -100,6 +101,22 @@ impl ActionHandler {
                                 continue;
                             }
                             let span = message.spans[0].rls_span().zero_indexed();
+
+                            // build a more sophisticated error message
+                            let mut message_text = message.message.clone();
+                            for sp in &message.spans {
+                                if sp.is_primary && sp.label.is_some() {
+                                    message_text.push_str("\n⇒ ");
```

Comment:
VS Code presents all error messages in a single line in the problems view. To show where those messages are joined I wanted to join them with another symbol. The problem with `->` and `=>` is that they both are valid tokens in Rust. I prefer to use a separator symbol which is not part of the Rust language and thus cannot occur in any error message.
All editors which use RLS should be able to handle unicode anyway, since UTF-8 is the only allowed encoding.